aapcs64-morello/aapcs64-morello.rst

```diff
@@ -0,0 +1,858 @@
+    +------------------+--------------------------+------------------------------------------------------------------------+
+    | AAPCS64-morello  | This document            | Morello extensions to PCS for the Arm® 64-bit Architecture (AArch64).  |
+    +------------------+--------------------------+------------------------------------------------------------------------+
+    | AAPCS64_         | IHI 005D                 | Procedure Call Standard for the Arm 64-bit Architecture.               |
+    +------------------+--------------------------+------------------------------------------------------------------------+
+
+Terms and Abbreviations
+-----------------------
+
+The ABI for the Arm 64-bit Architecture uses the following terms and abbreviations in addition
+to the terms and abbreviations used in the AAPCS64_ document.
+
+AAPCS64-cap
+    The pure capability Procedure Call Standard for the Arm 64-bit
+    Architecture (AArch64)
+
+C64
+    Execution state where PSTATE.C64 is set.
+
+A64
+    Execution state where PSTATE.C64 is cleared.
+
+Capability
+    The capability data type is an unforgeable token of authority which provides
+    a foundation for fine grained memory protection and strong compartmentalisation.
+
+Deriving a capability
+    A capability value CV2 is said to be derived from a capability value CV1
+    when CV2 is a copy of CV1 with optionally removed permissions and/or
+    optionally narrowed bounds (base increased or limit reduced).
+
+More specific terminology is defined when it is first used.
+
+Scope
+=====
+
+This document extends the AAPCS64 calling convention described in
+the AAPCS64_ document in order to add support for capabilities, and adds an
+additional Procedure Calling Standard (AAPCS64-cap). AAPCS64-cap is identical to AAPCS64,
+except for the differences documented here.
+
+Introduction
+============
+
+The AAPCS64 is the first revision of Procedure Call standard for the Arm 64-bit Architecture. It forms part of the complete ABI specification for the Arm 64-bit Architecture.
+
+The AAPCS64-cap is a Procedure Call standard for the Arm 64-bit Architecture designed to implement a software environment where all memory accesses are performed using capabilities.
+
+Design Goals
+------------
+
+AAPCS64 and AAPCS64-cap have the same design goals as described in the AAPCS64_ document.
+
+An additional design goal for the capability-aware AAPCS64 is to support interworking with legacy AAPCS64 code, following only the Procedure Calling Standard described in the AAPCS64_                 document.
+
+Conformance
+-----------
+
+The AAPCS64 and AAPCS64-cap have the same conformance rules as defined in AAPCS64_, with the additional requirement that:
+
+- At each call where the control transfer instruction is subject to a BL-type relocation at static link time, rules on the use of r16 and r17 (CIP0, CIP1, IP0, IP1) are observed (`Design Goals`_).
```

Comment:
I think this links to the wrong section. In the original document this links to "Use of CIP0 and CIP1 by the linker"